dashboard/config/routes.rb

```diff
@@ -13,6 +13,12 @@
 
   patch '/api/v1/user_scripts/:script_id', to: 'api/v1/user_scripts#update'
 
+  patch '/api/v1/user_school_infos/:id/update_last_confirmation_date', to: 'user_school_infos#update_last_confirmation_date'
+
+  patch '/api/v1/user_school_infos/:id/update_end_date', to: 'user_school_infos#update_end_date'
+
+  patch '/api/v1/user_school_infos/:id/update_school_info_id', to: 'user_school_infos#update_school_info_id'
```

Comment:
these (and also the existing `user_scripts#update` route) should go in the `api/v1` namespace: https://github.com/code-dot-org/code-dot-org/blob/c6fa999186736e517dafe8e6d2787920764b6338/dashboard/config/routes.rb#L585